reV/bespoke/bespoke.py

```diff
@@ -1217,6 +1218,34 @@ def meta(self):
             meta = meta[0]
         return meta
 
+    def sam_sys_inputs_with_site_data(self, gid):
+        """Update the sam_sys_inputs with site data for the given GID.
+
+        Site data is extracted from the project points DataFrame. Every
+        column in the project DataFrame becomes a key in the site_data
+        output dictionary.
+
+        Parameters
+        ----------
+        gid : int
+            SC point gid for site to pull site data for.
+
+        Returns
+        -------
+        dictionary : dict
+            SAM system config with extra keys from the project points
+            DataFrame.
+        """
+
+        gid_idx = self._project_points.index(gid)
+        site_data = self._project_points.df.iloc[gid_idx]
+
+        site_sys_inputs = self._project_points[gid][1]
+        site_sys_inputs.update({k: v for k, v in site_data.to_dict().items()
+                                if not isinstance(v, float)
+                                or not np.isnan(v)})
```

Comment:
This logic is confusing but its basically don't set if float and NaN right? I always prefer writing it how you would say it so like `if not (float and NaN)`

Reply:
Fair enough, I'll update that